src/test/HostActivationTests/FrameworkResolution/ApplyPatchesSettings.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.Cli.Build;
+using Microsoft.DotNet.Cli.Build.Framework;
+using System;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.FrameworkResolution
+{
+    public class ApplyPatchesSettings :
+        FrameworkResolutionBase,
+        IClassFixture<ApplyPatchesSettings.SharedTestState>
+    {
+        private SharedTestState SharedState { get; }
+
+        public ApplyPatchesSettings(SharedTestState sharedState)
+        {
+            SharedState = sharedState;
+        }
+
+        [Fact]
+        public void Default()
+        {
+            RunTest(
+                runtimeConfig => runtimeConfig
+                    .WithFramework(MicrosoftNETCoreApp, "5.1.2"),
+                result => result.Should().Pass()
+                    .And.HaveResolvedFrameworkVersion("5.1.3"));
+        }
+
+        [Fact]
+        public void RuntimeConfigOnly()
+        {
+            RunTest(
+                runtimeConfig => runtimeConfig
+                    .WithFramework(MicrosoftNETCoreApp, "5.1.2")
+                    .WithApplyPatches(false),
+                result => result.Should().Pass()
+                    .And.HaveResolvedFrameworkVersion("5.1.2"));
+        }
+
+        [Fact]
+        public void FrameworkReferenceOnly()
+        {
+            RunTest(
+                runtimeConfig => runtimeConfig
+                    .WithFramework(new RuntimeConfig.Framework(MicrosoftNETCoreApp, "5.1.2")
+                        .WithApplyPatches(false)),
+                result => result.Should().Pass()
+                    .And.HaveResolvedFrameworkVersion("5.1.2"));
+        }
+
+        [Fact]
+        public void Priority()
+        {
+            RunTest(
+                runtimeConfig => runtimeConfig
+                    .WithFramework(new RuntimeConfig.Framework(MicrosoftNETCoreApp, "5.1.2")
+                        .WithApplyPatches(false))
+                    .WithApplyPatches(true),
+                result => result.Should().Pass()
+                    .And.HaveResolvedFrameworkVersion("5.1.2"));
```

Comment:
Might be a better test if we added the framework name MicrosoftNETCoreApp (and "MiddleWare" in later tests)
```
.And.HaveResolvedFrameworkVersion(MicrosoftNETCoreApp, "5.1.2"));
```

Reply:
Good point - modified the mock to write this in a better format and added the stricter validation everywhere.